server/config/parameters.go

```diff
@@ -0,0 +1,218 @@
+func (p *Parameter) IsReadOnly() bool {
+	switch strings.ToLower(p.Name) {
+	case "server_version", "server_encoding", "lc_collate", "lc_ctype", "is_superuser":
+		return true
+	}
+	switch p.Context {
+	case ParameterContextInternal, ParameterContextPostmaster, ParameterContextSighup,
+		ParameterContextSuperUserBackend, ParameterContextBackend:
+		// Read the docs above the ParameterContext
+		// TODO: some of above contexts need support, return error for now
+		return true
+	case ParameterContextSuperUser, ParameterContextUser:
+		// TODO: need to check for 'superuser' and appropriate 'SET' privileges.
+		// Can be set from `postgresql.conf`, or within a session via the `SET` command.
+		return false
+	}
+	return false
+}
+
+// IsGlobalOnly implements SystemVariable.
+func (p *Parameter) IsGlobalOnly() bool {
+	// TODO: fix - postgres SESSION parameters are considered as GLOBAL in gms for now.
+	return true
+}
+
+// GetNotifyChanged implements SystemVariable.
+func (p *Parameter) GetNotifyChanged() func(sql.SystemVariableScope, sql.SystemVarValue) error {
+	// TODO: fix - some parameters might need them, but return nil for now.
+	return nil
+}
+
+// ParameterContext sets level of difficulty of changing the parameter settings.
+// For more detailed description on how to change the settings of specific context,
+// https://www.postgresql.org/docs/current/view-pg-settings.html
+type ParameterContext string
+
+// The following constants are in order of decreasing difficulty of changing the setting.
+const (
+	ParameterContextInternal         ParameterContext = "internal"
+	ParameterContextPostmaster       ParameterContext = "postmaster"
+	ParameterContextSighup           ParameterContext = "sighup"
+	ParameterContextSuperUserBackend ParameterContext = "superuser-backend"
+	ParameterContextBackend          ParameterContext = "backend"
+	ParameterContextSuperUser        ParameterContext = "superuser"
+	ParameterContextUser             ParameterContext = "user"
+)
+
+// ParameterSource sets the source of the current parameter value.
+type ParameterSource string
+
+const (
+	ParameterSourceClient ParameterSource = "client"
+	// ParameterSourceConfigurationFile means that the parameter needs to set
+	// its Default and ResetVal to what's defined in the given config file.
+	ParameterSourceConfigurationFile ParameterSource = "configuration file"
+	ParameterSourceDefault           ParameterSource = "default"
+	// ParameterSourceOverride means the default and reset value needs to be set at server start time
+	// TODO: currently the default and reset values are dummy values.
+	ParameterSourceOverride ParameterSource = "override"
+)
+
+// offsetRegex is a regex for matching MySQL offsets (e.g. +01:00).
+var offsetRegex = regexp.MustCompile(`(?m)^([+\-])(\d{2}):(\d{2})$`)
+
+// MySQLOffsetToDuration takes in a MySQL timezone offset (e.g. "+01:00") and returns it as a time.Duration.
+// If any problems are encountered, an error is returned.
```

Comment:
Perhaps a note on why we're using `MySQL` offsets, which I'm assuming is because it's what the GMS types require.

Reply:
Ah this isn't MySQL specific, I copied it from GMS as it was in `internal` package. Postgres allows setting `TimeZone` to an offset value. This method parses a valid offset. I'll update its name and description.